tutorial3.md

```diff
@@ -0,0 +1,419 @@
+
+```yaml
+command:
+  - etcd
+  - --name=k8s-demo
+  - --trusted-ca-file=/etc/kubernetes/pki/etcd/ca.crt
+  - --key-file=/etc/kubernetes/pki/etcd/server.key
+  - --cert-file=/etc/kubernetes/pki/etcd/server.crt
+  - --client-cert-auth=true
+  - --listen-client-urls=https://0.0.0.0:2379
+  - --advertise-client-urls=https://localhost:2379
+```
+
+We set up a data directory, so we may as well use it:
+
+```yaml
+command:
+  - etcd
+  - --name=k8s-demo
+  - --trusted-ca-file=/etc/kubernetes/pki/etcd/ca.crt
+  - --key-file=/etc/kubernetes/pki/etcd/server.key
+  - --cert-file=/etc/kubernetes/pki/etcd/server.crt
+  - --client-cert-auth=true
+  - --listen-client-urls=https://0.0.0.0:2379
+  - --advertise-client-urls=https://localhost:2379
+  - --data-dir=/var/lib/etcd
+```
+
+## Putting it all together
+
+Add our command to our container:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    component: etcd
+    tier: control-plane
+  name: etcd
+  namespace: kube-system
+spec:
+  containers:
+  - image: k8s.gcr.io/etcd:3.2.24
+    name: etcd
+    command:
+    - etcd
+    - --name=k8s-demo
+    - --trusted-ca-file=/etc/kubernetes/pki/etcd/ca.crt
+    - --key-file=/etc/kubernetes/pki/etcd/server.key
+    - --cert-file=/etc/kubernetes/pki/etcd/server.crt
+    - --client-cert-auth=true
+    - --listen-client-urls=https://0.0.0.0:2379
+    - --advertise-client-urls=https://localhost:2379
+    - --data-dir=/var/lib/etcd
+    volumeMounts:
+    - mountPath: /var/lib/etcd
+      name: etcd-data
+    - mountPath: /etc/kubernetes/pki/etcd
+      name: etcd-certs
+  hostNetwork: true
+  volumes:
+  - hostPath:
+      path: /var/lib/etcd
+      type: DirectoryOrCreate
+    name: etcd-data
+  - hostPath:
+      path: /etc/kubernetes/pki/etcd
+      type: Directory
+    name: etcd-certs
+```
+
+Now let's write it out and have `kubelet pick it up`.
+Save the above yaml in `/etc/kubernetes/pods` just like our example pod.
+
+And now it should be running! Let's see:
+
+```console
+$ sudo docker ps | grep etcd
+4dd1037c5b3b        3cab8e1b9802           "etcd --name=k8s-dem…"   50 seconds ago      Up 49 seconds                           k8s_etcd_etcd-k8s-tutorial_kube-system_9b90156916d8b089087c0
+c7aa8e7018d_0
+8750c1757738        k8s.gcr.io/pause:3.1   "/pause"                 51 seconds ago      Up 49 seconds                           k8s_POD_etcd-k8s-tutorial_kube-system_9b90156916d8b089087c0c
+7aa8e7018d_0
+```
+
+Awesome, it's running!
+
+## Health Check
+
+We can check if our etcd is healthy in the same way an external health checker would.
+First, to connect, we'll need valid certificates, signed by the etcd CA.
+kubeadm can do it for us:
+
+```console
+$ sudo kubeadm init phase certs etcd-healthcheck-client
+[certs] Generating "etcd/healthcheck-client" certificate and key
+```
+
+Once again, we can make sure this is signed:
+
+```console
+$ openssl verify -CAfile /etc/kubernetes/pki/etcd/ca.crt /etc/kubernetes/pki
```

Comment:
newline munged here, should be:
`$ openssl verify -CAfile /etc/kubernetes/pki/etcd/ca.crt /etc/kubernetes/pki/etcd/healthcheck-client
.crt`